Content.Server/GameObjects/Components/Weapon/Melee/StunbatonComponent.cs

```diff
@@ -0,0 +1,91 @@
+using System.Collections.Generic;
+using Content.Server.GameObjects.Components.Mobs;
+using Content.Server.GameObjects.EntitySystems;
+using Content.Shared.Audio;
+using Robust.Server.GameObjects;
+using Robust.Server.GameObjects.EntitySystems;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.Interfaces.Random;
+using Robust.Shared.IoC;
+using Robust.Shared.Random;
+using Robust.Shared.Serialization;
+using Robust.Shared.ViewVariables;
+
+namespace Content.Server.GameObjects.Components.Weapon.Melee
+{
+    [RegisterComponent]
+    public class StunbatonComponent : MeleeWeaponComponent, IUse
+    {
+        [Dependency] private IRobustRandom _robustRandom;
+        [Dependency] private IEntitySystemManager _entitySystemManager;
+
+        public override string Name => "Stunbaton";
+
+        private bool _activated = false;
+
+        [ViewVariables(VVAccess.ReadWrite)]
+        private float _paralyzeChance = 0.25f;
+
+        [ViewVariables(VVAccess.ReadWrite)]
+        private float _paralyzeTime = 10f;
+
+        [ViewVariables(VVAccess.ReadWrite)]
+        private float _slowdownTime = 5f;
+
+        [ViewVariables]
+        public bool Activated => _activated;
+
+        public override void ExposeData(ObjectSerializer serializer)
+        {
+            base.ExposeData(serializer);
+
+            serializer.DataField(ref _paralyzeChance, "paralyzeChance", 0.25f);
+            serializer.DataField(ref _paralyzeTime, "paralyzeTime", 10f);
+            serializer.DataField(ref _slowdownTime, "slowdownTime", 5f);
+        }
+
+        public override void OnHitEntities(IEnumerable<IEntity> entities)
```

Comment:
Just overriding this means that the damage code still runs, doesn't it?

Reply:
Yeah, that's intended. I guess I could change the prototype to make this not make too much damage.